pyrealm/core/solar.py

```diff
@@ -87,3 +661,254 @@ def calc_heliocentric_longitudes(
+    """
+
+    declination = -23.4 * (np.pi / 180) * np.cos((2 * np.pi * (td + 10)) / 365)
+
+    return declination
+
+
+def local_hour_angle(t: NDArray, t0: NDArray) -> NDArray:
+    r"""Calculate the local hour angle for a given time and solar noon.
+
+    The local hour angle is a measure of time, expressed in angular terms, that
+    indicates the position of the sun relative to solar noon. This function calculates
+    the local hour angle by determining the difference between the current time (t)
+    and the solar noon time (t_0), and then converting this difference into an angle.
+
+    Equation implemented from A15 :citet:'de Pury and Farquhar:1997'.
+
+    .. math::
+        h = \pi \cdot \frac{t - t_0}{12}
+
+    Args:
+        t           : Array of current time values in hours (as a floating-point
+                        number).
+        t0          : Array of solar noon time values in hours (as a floating-point
+                        number).
+
+    Returns:
+        The local hour angle in radians (as a floating-point number array), which
+        represents the angular distance of the sun from the local meridian at the
+        given time.
+
+    """
+
+    h = np.pi * (t - t0) / 12
+
+    return h
+
+
+def solar_noon(L_e: float, L_s: float, E_t: NDArray) -> NDArray:
+    r"""Calculate the solar noon time for a given location.
+
+    The solar noon is the time of day when the sun is at its highest point in the sky
+    for a given location.
+    This function calculates the solar noon by adjusting the standard noon time
+    (12:00 PM) based on the difference between the local longitude (L_e) and the
+    standard meridian (L_s) and the equation of time (E_t). Based on EqA16,
+    :cite:t:'de Pury and Farquhar:1997'.
+
+    .. math::
+        t_0 = 12 + \frac{4 \cdot -(L_e - L_s) - E_t}{60}
+
+    Args:
+        L_e         : Local longitude of the observer in degrees (positive for east,
+                        negative for west).
+        L_s         : Longitude of the standard meridian for the observer's time zone
+                        in degrees.
+        E_t         : Equation of time in minutes, accounting for the irregularity of
+                        the Earth's orbit and axial tilt.
+
+    Returns:
+        The solar noon time in hours (as a floating-point number), which can be
+        interpreted as a time of day.
+
+    """
+
+    t0 = 12 + (4 * -(L_e - L_s) - E_t) / 60
+
+    return t0
+
+
+def equation_of_time(day_angle: NDArray) -> NDArray:
+    r"""Calculates equation of time in minutes.
+
+    Based on eqn 1.4.1 :cite:t:'Iqbal:1983' rather than A17
+    :cite:t:'de Pury and Farquahar:1997' due to incorrect reported implementation in
+    the latter.
+
+    .. math::
+
+    E_t = \left( 0.000075
+    + 0.001868 \cdot \cos(\Gamma)
+    - 0.032077 \cdot \sin(\Gamma)
+    - 0.014615 \cdot \cos(2\Gamma)
+    - 0.04089 \cdot \sin(2\Gamma) \right)
+    \times 229.18
```

Comment:
Needs to be indented to render as maths

```suggestion
    .. math::

        E_t = \left( 0.000075
            + 0.001868 \cdot \cos(\Gamma)
            - 0.032077 \cdot \sin(\Gamma)
            - 0.014615 \cdot \cos(2\Gamma)
            - 0.04089 \cdot \sin(2\Gamma) \right)
            \times 229.18
```